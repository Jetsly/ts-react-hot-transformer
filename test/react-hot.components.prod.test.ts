import * as ts from 'typescript';
import { join } from 'path';
import { readdirSync, readFileSync, statSync } from 'fs';
import transformerFactory from '../src/react-hot.dev';
import transformerProdFactory from '../src/react-hot.prod';
const printer = ts.createPrinter();
const FIXTURES_DIR = join(__dirname, '__fixtures__');
function addRHLPlugin(prod = false) {
  return prod ? transformerProdFactory : transformerFactory;
}

describe('React components', () => {
  const fixturesDir = join(FIXTURES_DIR, 'drop-hot');
  readdirSync(fixturesDir).forEach(fixtureName => {
    const fixtureFile = join(fixturesDir, fixtureName);
    // const testFile: Array<string> = ['bindings.js'];
    // if (statSync(fixtureFile).isFile() && testFile.indexOf(fixtureName) > -1) {
    if (statSync(fixtureFile).isFile()) {
      it(fixtureName.split('-').join(' '), () => {
        const sourceCode = readFileSync(fixtureFile, 'utf-8');
        const source = ts.createSourceFile(fixtureName, sourceCode, ts.ScriptTarget.ES2016, true);
        const result = ts.transform(source, [addRHLPlugin(true)]);
        const transformedSourceFile = result.transformed[0];
        const resultCode = printer.printFile(transformedSourceFile);
        expect(resultCode).toMatchSnapshot();
      });
    }
  });
});
