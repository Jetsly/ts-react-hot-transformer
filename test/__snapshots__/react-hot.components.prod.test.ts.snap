// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React components drop hot.js 1`] = `
"import React from 'react';
import { hot } from 'react-hot-loader/root';
import { hot as namedHot, foo as namedFoo } from 'react-hot-loader/root';
import { hot as namedHot2 } from 'react-hot-loader';
import { hot as notRHLHot } from 'not-react-hot-loader';
import * as RHL from 'react-hot-loader/root';
import * as RHL2 from 'react-hot-loader/root';
import * as NOTRHL from 'not-react-hot-loader';
const App = () => <div>Hello World!</div>;
const a = App;
const c = App;
const d = App;
const e = App;
foo(App);
notRHLHot(App);
namedFoo(App);
RHL.foo(App);
NOTRHL.hot(App);
export { a, b, c, d, e };
(function (namedHot2) {
    const b = namedHot(App);
    const c = namedHot2(module)(App);
})(1);
"
`;
